### PR TITLE
Add flexible-release-model doc

### PR DESCRIPTION
**Pull request by matyasselmeci**
Based on the "New Release Model" Google Doc

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -40,6 +40,7 @@ pages:
     - HTCondor Prerelease Testing: 'release/condor-itb-testing.md'
     - Empty Packages: 'release/empty-pkgs.md'
     - Acceptance Testing: 'release/acceptance-testing.md'
+    - Flexible Release Model: 'release/flexible-release-model.md'
```
Comment (brianhlin) [inline mkdocs.yml]: Toss this into the 'Policy' section

**Comment by brianhlin** [comment]
@timtheisen you should make sure to look this document over to verify that it's satisfactory

**Comment by timtheisen** [comment]
The reads like a proposal, not a policy. Reading this in January, you would be confused. Change the "Rationale" to "Introduction" and make it refer to the past. Also, change "Proposal" to "Release Model" and make it refer to the present.

**Comment by bbockelm** [comment]
@timtheisen merge already occurred yesterday.  For follow-ups, please open a new PR.

**Pull request merged**